### PR TITLE
make the code neat and clean without global variables

### PR DESCRIPTION
**Pull request by kvedala**
please review how I updated the code

### CODE REVIEW

File: backtracking/graph_coloring.cpp [hunk 1]
```diff
@@ -1,18 +1,36 @@
+/**
+ * @file
+ * @brief brief one line description here
+ */
 #include <iostream>
 
-/*
- * Number of vertices in the graph
+// created a namespace to ensure all globals and functions are
+// restricted to this file
```
Comment (Panquesito7) [inline backtracking/graph_coloring.cpp:8]: Please use a comment block.

File: backtracking/graph_coloring.cpp [hunk 2]
```diff
@@ -1,18 +1,36 @@
...
+ * @returns `true` if ....
+ * @returns `false` if ....
  */
-void printSolution(int color[]);
-
-/* A utility function to check if the current color assignment
-   is safe for vertex v */
-bool isSafe(int v, bool graph[V][V], int color[], int c) {
+template <int V>
+bool isSafe(int v, const bool graph[V][V], const int *color, int c) {
```
Comment (Panquesito7) [inline backtracking/graph_coloring.cpp:33]: Will LGTM consider this as a short variable name?

File: backtracking/graph_coloring.cpp (hunk 1 above)
Comment (kvedala) [reply backtracking/graph_coloring.cpp:8]: Dude, this is for you. If I fix the whole code, I will pull to the master branch myself. I am trying to do you a favour and teach you.

File: backtracking/graph_coloring.cpp (hunk 2 above)
Comment (kvedala) [reply backtracking/graph_coloring.cpp:33]: The variable name is local to the function and not a global variable. If you READ the LGTM reports, they are regarding *global* variables

File: backtracking/graph_coloring.cpp (hunk 1 above)
Comment (Panquesito7) [reply backtracking/graph_coloring.cpp:8]: Sorry about that.
Thanks for this PR, though.

Comment (kvedala) [reply backtracking/graph_coloring.cpp:8]: study the fixes I made and how I navigated through your questions.. you wanted to know how to get rid of global variables. How did I do it in this code?
How did I add related documetnation, etc.. Would help you to make better contributions and make better fixes.

**Comment by Panquesito7** [comment]
Thanks, @kvedala! 🙂

**Comment by kvedala** [comment]
> Thanks, @kvedala! 🙂

You are welcome :)

**Pull request merged**